python/lwt_interface/tskit_lwt_interface.h

```diff
@@ -1458,72 +1458,205 @@ parse_table_collection_dict(tsk_table_collection_t *tables, PyObject *tables_dic
     return ret;
 }
 
+typedef struct _tsklwt_table_col_t {
+    const char *name;
+    void *data;
+    npy_intp num_rows;
+    int type;
+} tsklwt_table_col_t;
+
+typedef struct _tsklwt_ragged_col_t {
+    const char *name;
+    void *data;
+    tsk_size_t *offset;
+    npy_intp num_rows;
+    npy_intp data_len;
+    int type;
+} tsklwt_ragged_col_t;
+
+typedef struct _tsklwt_table_desc_t {
+    const char *name;
+    tsklwt_table_col_t *cols;
+    tsklwt_ragged_col_t *ragged_cols;
+    char *metadata_schema;
+    tsk_size_t metadata_schema_length;
+} tsklwt_table_desc_t;
+
 static int
-write_table_arrays(tsk_table_collection_t *tables, PyObject *dict)
+write_table_col(const tsklwt_table_col_t *col, PyObject *table_dict)
+{
+    int ret = -1;
+
+    PyArrayObject *array
+        = (PyArrayObject *) PyArray_EMPTY(1, &col->num_rows, col->type, 0);
+    if (array == NULL) {
+        goto out;
+    }
+    memcpy(PyArray_DATA(array), col->data, col->num_rows * PyArray_ITEMSIZE(array));
+    if (PyDict_SetItemString(table_dict, col->name, (PyObject *) array) != 0) {
+        goto out;
+    }
+    ret = 0;
+out:
+    Py_XDECREF(array);
+    return ret;
+}
+
+static int
+write_ragged_col(
+    const tsklwt_ragged_col_t *col, PyObject *table_dict, bool force_offset_64)
+{
+    int ret = -1;
+    char offset_col_name[128];
+    npy_intp offset_len = col->num_rows + 1;
+    PyArrayObject *data_array = NULL;
+    PyArrayObject *offset_array = NULL;
+    bool offset_64 = force_offset_64; // || col->offset[col->num_rows] > UINT32_MAX
+    int offset_type = offset_64 ? NPY_UINT64 : NPY_UINT32;
+    /* TODO change this to 32 bit when we flip tsk_size_t over */
```

Comment:
By this comment you mean the whole logic of this section will be flipped? I think I get it!

Reply:
Yeah, exactly.